src/Eppie.App/Eppie.App.ViewModels/DecentralizedAccountSettingsPageViewModel.cs

```diff
@@ -388,27 +314,86 @@ private async void OnAccountSettingsModelPropertyChanged(object sender, System.C
                 finally
                 {
                     _accountSettingsModel.IsNetworkLocked = false;
+                    ClaimNameCommand?.NotifyCanExecuteChanged();
                 }
             }
         }
 
-        public void Dispose()
+        private async Task ClaimNameAsync()
         {
-            Dispose(true);
-            GC.SuppressFinalize(this);
+            if (AccountSettingsModel == null || !AccountSettingsModel.IsSenderNameVisible)
```

Comment:
you can use `AccountSettingsModel?.IsSenderNameVisible != true` or `AccountSettingsModel is null || ...`